source/control/compact-combinator-surface.lua

```diff
@@ -15,25 +15,51 @@ local private = {} -- private methods
 
 
 -- returns the coordinates of a new chunk to be used for a compact-combinator
-function Surface.newSpot(x, y)
+function Surface.newSpot(pos_x, pos_y)
 	local data = private.data()
 	
-	local c = { math.floor(x/32), math.floor(y/32) }
-	if not private.isChunkUsed(c) then
-		private.markChunk(c, true)
-		return c
-	end
-	
-	for x=-1,1 do for y=-1,1 do
-		if math.abs(x)+math.abs(y) == 1 then
-			local c2 = {c[1]+x, c[2]+y}
-			if not private.isChunkUsed(c2) then
-				private.markChunk(c2, true)
-				return c2
+	-- spiral search
+	local x = math.floor(pos_x / 32)
+	local y = math.floor(pos_y / 32)
+	local radius = 1
+	local direction = 1
+
+	while true do
+		-- move Y up/down
+		for dy = 0,radius do
```

Comment:
note radius is inclusive, so it starts with 2 iterations (dy=0, dy=1), I changed it to dy=1,radius such that it starts with minimal spiral radius of 1.